plugins/org.eclipse.fordiac.ide.model/src/org/eclipse/fordiac/ide/model/resource/FordiacTypeResource.java

```diff
@@ -116,21 +132,37 @@ protected void doSave(final OutputStream outputStream, final Map<?, ?> options)
 					MessageFormat.format(Messages.FordiacTypeResource_SaveToUnsupportedURI, uri.toString()));
 		}
 		try {
-			final var fbtFile = ResourcesPlugin.getWorkspace().getRoot()
-					.getFile(new Path(this.uri.toPlatformString(true)));
-			var typeEntryForFile = TypeLibraryManager.INSTANCE.getTypeEntryForFile(fbtFile);
-			if (typeEntryForFile == null) {
-				typeEntryForFile = TypeLibraryManager.INSTANCE.getTypeLibrary(fbtFile.getProject())
-						.createTypeEntry(fbtFile);
-			}
-			typeEntryForFile.setTypeEditable(EcoreUtil.copy((LibraryElement) getContents().get(0)));
-			typeEntryForFile.setLastModificationTimestamp(typeEntryForFile.getFile().getModificationStamp());
-			AbstractTypeExporter.saveType(typeEntryForFile, outputStream);
+			saveType(getContents().get(0), outputStream);
 		} catch (final Exception e) {
 			throw new IOWrappedException(e);
 		}
 	}
 
+	public static void saveType(final EObject le, final OutputStream outputStream) {
+		final AbstractTypeExporter exporter = getTypeExporter(le);
+		if (exporter != null) {
+			try (InputStream inputStream = exporter.getFileContent()) {
+				inputStream.transferTo(outputStream);
+			} catch (final IOException e) {
+				FordiacLogHelper.logError(e.getMessage(), e);
```

Comment:
Please throw the exception from `doSave()` instead of silently logging it to allow proper error handling in the caller.

Reply:
ok will do that. Another advantage of my clean-up this was hidden in the AbstractTypeExporter bevore.

Reply:
Yes, I very much appreciate the clean-up and am glad we noticed this now.